src/platforms/common/stm32/traceswoasync.c

```diff
@@ -41,10 +43,31 @@
 #include <libopencm3/stm32/usart.h>
 #include <libopencm3/stm32/dma.h>
 
+#if defined(DMA_STREAM0)
+#define dma_channel_reset(dma, channel)   dma_stream_reset(dma, channel)
+#define dma_enable_channel(dma, channel)  dma_enable_stream(dma, channel)
+#define dma_disable_channel(dma, channel) dma_disable_stream(dma, channel)
+
+#define DMA_PSIZE_8BIT DMA_SxCR_PSIZE_8BIT
+#define DMA_MSIZE_8BIT DMA_SxCR_MSIZE_8BIT
+#define DMA_PL_HIGH    DMA_SxCR_PL_HIGH
+#else
+#define DMA_PSIZE_8BIT DMA_CCR_PSIZE_8BIT
+#define DMA_MSIZE_8BIT DMA_CCR_MSIZE_8BIT
+#define DMA_PL_HIGH    DMA_CCR_PL_HIGH
+#endif
+
+#define TRACESWOASYNC_ALLOCS
```

Comment:
This should be determined by the platform header, not in this file. Having this alternative mode is certainly quite interesting, but it should not be the default for all platforms we think. We're also of the opinion that if init allocates, deinit must deallocate in this paradigm, otherwise that's a permanent memory usage ever after turning Trace/SWO capture back off, eliminating the benefit, particularly if the probe is then used to scan a large JTAG or SWD chain after.

(Perhaps this should be enabled on F1 based platforms, but not the F4's? for example)